src/main/java/io/cdap/plugin/google/common/GoogleAuthBaseConfig.java

```diff
@@ -229,27 +243,48 @@ private boolean validateServiceAccount(FailureCollector collector) {
     return collector.getValidationFailures().size() == 0;
   }
 
-  private void validateCredentials(FailureCollector collector, GoogleDriveClient driveClient) throws IOException {
-    try {
-      driveClient.checkRootFolder();
-    } catch (GoogleJsonResponseException e) {
-      collector.addFailure(e.getDetails().getMessage(), "Provide valid credentials.")
-        .withConfigProperty(NAME_SERVICE_ACCOUNT_TYPE)
-        .withStacktrace(e.getStackTrace());
-    }
-  }
-
-  private void validateDirectoryIdentifier(FailureCollector collector, GoogleDriveClient driveClient)
+  private boolean isDirectoryOrFileAccessible(FailureCollector collector, GoogleDriveClient driveClient)
     throws IOException {
-    if (!containsMacro(DIRECTORY_IDENTIFIER)) {
+
+    if (getIdentifierType().equals(IdentifierType.DIRECTORY_IDENTIFIER) &&
+      directoryIdentifier != null && !containsMacro(DIRECTORY_IDENTIFIER)) {
       try {
         driveClient.isFolderAccessible(directoryIdentifier);
+        return true;
       } catch (GoogleJsonResponseException e) {
         collector.addFailure(e.getDetails().getMessage(), "Provide an existing folder identifier.")
           .withConfigProperty(DIRECTORY_IDENTIFIER)
           .withStacktrace(e.getStackTrace());
       }
     }
+
+    if (getIdentifierType().equals(IdentifierType.FILE_IDENTIFIER) &&
+      fileIdentifier != null && !containsMacro(FILE_IDENTIFIER)) {
+      try {
+        driveClient.isFileAccessible(fileIdentifier);
+        return true;
+      } catch (GoogleJsonResponseException e) {
+        collector.addFailure(e.getDetails().getMessage(), "Provide an existing file identifier.")
+          .withConfigProperty(FILE_IDENTIFIER)
+          .withStacktrace(e.getStackTrace());
+      }
+    }
+    return false;
+  }
+
+  protected void checkIfDirectoryOrFileIdentifierExists(FailureCollector collector) {
+    if (getIdentifierType().equals(IdentifierType.DIRECTORY_IDENTIFIER) && directoryIdentifier == null
```

Comment:
use `Strings.isNullOrEmpty` instead of `directoryIdentifier == null`.

similar comment applies to whole of the PR.